core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -803,8 +804,10 @@ class CivilizationInfo {
             placedUnit.religion = 
                 if (city != null) city.cityConstructions.cityInfo.religion.getMajorityReligionName()
                 else religionManager.religion?.name
-            if (placedUnit.hasUnique("Can spread religion [] times"))
-                placedUnit.abilityUsedCount["Religion Spread"] = 0
+            if (placedUnit.canDoReligiousAction(Constants.spreadReligionAbilityCount))
+                placedUnit.abilityUsedCount[Constants.spreadReligionAbilityCount] = 0
+            if (placedUnit.canDoReligiousAction(Constants.removeHeresyAbilityCount))
+                placedUnit.abilityUsedCount[Constants.removeHeresyAbilityCount] = 0
```

Comment:
for(unique in placedUnit.getUniques("Can [] [] times) placedUnit.abilityUsedCount[unique.params[0]] = 0